src/java/org/apache/cassandra/io/sstable/SSTableWatcher.java

```diff
@@ -53,4 +53,11 @@ default Set<Component> discoverComponents(Descriptor descriptor, Set<Component>
     {
         return existing;
     }
+
+    /**
+     * Called before executing index rebuild on given sstable
+     */
+    default void onIndexRebuild(Descriptor descriptor)
```

Comment:
Nit: technically "rebuild" is maybe a tad imprecise as this is call on the initial build of an index too. Essentially this is called every time we build an index on an existing sstable, so any index build that isn't at flush time. I'd at least add the precision to the javadoc, but renaming to `onIndexBuild` might be a tad better? I might also suggest passing the `SSTableReader` instead of the `Descriptor` as a way to imply that this is call on existing sstable, and so not flush time, but maybe I'm reading too much into this :)

Reply:
good point, updated